### PR TITLE
Allow not to create subdirectory for jdk extraction

### PR DESCRIPTION
**Pull request by EzzhevNikita**
**Task name**: JavaToolInstallerV0

**Description**: Added additional input for the task that allows disabling adding additional directory in jdkDestinationDirectory.
* Previously when task installed JDk into the directory that looks similar to this like (it created additional directory for JDK inside DestinationDirectory):
jdkDestinationDirectory/JAVA_HOME_8_X64_OpenJDK_zip

* In new behavior, by default, we don't change the installation path and it will look in the same format, but we could disable the creation of subdirectory, using createExtractDirectory input.

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) https://github.com/microsoft/azure-pipelines-tasks/issues/14778

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/JavaToolInstallerV0/javatoolinstaller.ts [hunk 1]
```diff
@@ -128,9 +128,15 @@ async function installJDK(sourceFile: string, fileExtension: string, archiveExtr
         jdkDirectory = await installPkg(sourceFile, extendedJavaHome, versionSpec);
     }
     else {
+        const createExtractDirectory: boolean = taskLib.getBoolInput('cleanDestinationDirectory', false);
+        let extractionDirectory: string = "";
+        if (createExtractDirectory){
```
Comment (DaniilShmelev) [inline Tasks/JavaToolInstallerV0/javatoolinstaller.ts:133]:
```suggestion
        if (createExtractDirectory) {
```
NIT

File: Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -21,9 +21,11 @@
   "loc.input.label.azureCommonVirtualFile": "Common virtual path",
   "loc.input.help.azureCommonVirtualFile": "Path to the JDK inside the Azure storage container.",
   "loc.input.label.jdkDestinationDirectory": "Destination directory",
-  "loc.input.help.jdkDestinationDirectory": "On Linux and Windows, this is used as the destination directory for JDK installation. On macOS, this directory is used as a temporary folder for extracting .dmg's since macOS doesn't support installing of JDK to specific directory.",
+  "loc.input.help.jdkDestinationDirectory": "On Linux and Windows, this is used as the destination directory for JDK installation. On macOS, this directory is used as a temporary folder for extracting of .dmg's since macOS doesn't support installing of JDK to specific directory.",
```
Comment (DaniilShmelev) [inline Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson:24]: Is this intentional? :)

File: Tasks/JavaToolInstallerV0/javatoolinstaller.ts [hunk 3]
```diff
@@ -128,9 +128,15 @@ async function installJDK(sourceFile: string, fileExtension: string, archiveExtr
         jdkDirectory = await installPkg(sourceFile, extendedJavaHome, versionSpec);
     }
     else {
+        const createExtractDirectory: boolean = taskLib.getBoolInput('cleanDestinationDirectory', false);
```
Comment (DaniilShmelev) [inline Tasks/JavaToolInstallerV0/javatoolinstaller.ts:131]:
```suggestion
        const createExtractDirectory: boolean = taskLib.getBoolInput('createExtractDirectory', false);
```

**Review by DaniilShmelev — APPROVED** [review]
Take a look at the comments, but other than that LGTM, thanks!
Let me test this manually as well

File: Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson (hunk 2 above)
Comment (EzzhevNikita) [reply Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson:24]: It is autogenerated file, so seems like when these changes introduced initially in task.json this file wasn't regenerated.

File: Tasks/JavaToolInstallerV0/javatoolinstaller.ts (hunk 1 above)
Comment (EzzhevNikita) [reply Tasks/JavaToolInstallerV0/javatoolinstaller.ts]: Thanks, fixed

File: Tasks/JavaToolInstallerV0/javatoolinstaller.ts (hunk 3 above)
Comment (EzzhevNikita) [reply Tasks/JavaToolInstallerV0/javatoolinstaller.ts]: Fixed

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Comment by DaniilShmelev** [comment]
Task failed with the following error:
`##[error]JDK file is not valid. Verify if JDK file contains only one root folder with 'bin' inside.`

Repro:
```yaml
- task: JavaToolInstallerM@0
  displayName: 'createExtractDirectory=false'
  inputs:
    versionSpec: 7
    jdkArchitectureOption: x64
    jdkSourceOption: AzureStorage
    azureResourceManagerEndpoint: $(StorageAccountConnection)
    azureStorageAccountName: $(StorageAccountName)
    azureContainerName: $(AzureContainerName)
    azureCommonVirtualFile: $(ArchiveName)
    jdkDestinationDirectory: ./tempAzure/java_dir
    createExtractDirectory: false
```

**Comment by EzzhevNikita** [comment]
> Task failed with the following error:
> `##[error]JDK file is not valid. Verify if JDK file contains only one root folder with 'bin' inside.`
> 
> Repro:
> 
> ```yaml
> - task: JavaToolInstallerM@0
>   displayName: 'createExtractDirectory=false'
>   inputs:
>     versionSpec: 7
>     jdkArchitectureOption: x64
>     jdkSourceOption: AzureStorage
>     azureResourceManagerEndpoint: $(StorageAccountConnection)
>     azureStorageAccountName: $(StorageAccountName)
>     azureContainerName: $(AzureContainerName)
>     azureCommonVirtualFile: $(ArchiveName)
>     jdkDestinationDirectory: ./tempAzure/java_dir
>     createExtractDirectory: false
> ```

Yeah thanks, I found this problem too

**Comment by DaniilShmelev** [comment]
Tested these changes manually with azure and local JDK. Found no issues.

**Pull request merged**